manticore/platforms/evm.py

```diff
@@ -1977,7 +2025,11 @@ def create_contract(self, price=0, address=None, caller=None, balance=0, init=No
         This is done when the byte code in the init byte array is actually run
         on the network.
         '''
-        address = self.create_account(address)
+        expected_address = self.create_account(self.new_address(sender=caller))
+        if address is None:
+            address = expected_address
+        elif caller is not None and address != expected_address:
+            raise EVMException("Error: contract created from address %x with nonce %d was expected to be at address %x, but create_contract was called with address=%x" % (caller, self.get_nonce(caller), expected_address, address))
```

Comment:
nit, EthereumError here, instead of EVMException? since this is actually an error

Reply:
Up until my second-to-last-commit, this file wasn't even importing `EthereumError`, yet it was using it in a few places. It's currently that way in master! Our tests just never triggered those throws, so we never experienced the `EthereumError is not defined` error. But you're right, it should really be an `EthereumError`.